File: backend/static/js/Position.js
class Position
{
    madeMoves = null;
    letters = null;
    possibleMoves = null;
    x = 0;
    constructor(mm, x_, letters_)
    {
        this.letters = letters_;
        this.x = x_;
        this.madeMoves = JSON.parse(JSON.stringify(mm));
        this.possibleMoves = this.createPossibleDirections();
        if (this.madeMoves.length < 15)
        {
            (this.madeMoves.push(this.x));
        }
    }
    createPossibleDirections()
    {
        var possibles = new Array();
        if (this.x == 0)
        {
            possibles.push(1);
            possibles.push(4);
            possibles.push(5);
        }
        else if (this.x == 1)
        {
            possibles.push(0);
            possibles.push(2);
            possibles.push(4);
            possibles.push(5);
            possibles.push(6);
        }
        else if (this.x == 2)
        {
            possibles.push(1);
            possibles.push(3);
            possibles.push(5);
            possibles.push(6);
            possibles.push(7);
        }
        else if (this.x == 3)
        {
            possibles.push(2);
            possibles.push(6);
            possibles.push(7);
        }
        else if (this.x == 4)
        {
            possibles.push(0);
            possibles.push(1);
            possibles.push(5);
            possibles.push(8);
            possibles.push(9);
        }
        else if (this.x == 5)
        {
            possibles.push(0);
            possibles.push(1);
            possibles.push(2);
            possibles.push(4);
            possibles.push(6);
            possibles.push(8);
            possibles.push(9);
            possibles.push(10);
        }
        else if (this.x == 6)
        {
            possibles.push(1);
            possibles.push(2);
            possibles.push(3);
            possibles.push(5);
            possibles.push(7);
            possibles.push(9);
            possibles.push(10);
            possibles.push(11);
        }
        else if (this.x == 7)
        {
            possibles.push(2);
            possibles.push(3);
            possibles.push(6);
            possibles.push(10);
            possibles.push(11);
        }
        else if (this.x == 8)
        {
            possibles.push(4);
            possibles.push(5);
            possibles.push(9);
            possibles.push(12);
            possibles.push(13);
        }
        else if (this.x == 9)
        {
            possibles.push(4);
            possibles.push(5);
            possibles.push(6);
            possibles.push(8);
            possibles.push(10);
            possibles.push(12);
            possibles.push(13);
            possibles.push(14);
        }
        else if (this.x == 10)
        {
            possibles.push(5);
            possibles.push(6);
            possibles.push(7);
            possibles.push(9);
            possibles.push(11);
            possibles.push(13);
            possibles.push(14);
            possibles.push(15);
        }
        else if (this.x == 11)
        {
            possibles.push(6);
            possibles.push(7);
            possibles.push(10);
            possibles.push(14);
            possibles.push(15);
        }
        else if (this.x == 12)
        {
            possibles.push(8);
            possibles.push(9);
            possibles.push(13);
        }
        else if (this.x == 13)
        {
            possibles.push(8);
            possibles.push(9);
            possibles.push(10);
            possibles.push(12);
            possibles.push(14);
        }
        else if (this.x == 14)
        {
            possibles.push(9);
            possibles.push(10);
            possibles.push(11);
            possibles.push(13);
            possibles.push(15);
        }
        else if (this.x == 15)
        {
            possibles.push(10);
            possibles.push(11);
            possibles.push(14);
        }
        var p = new Array();
        for (let i in possibles) {
            if(this.madeMoves.includes(i)){
                ;
            }else{
                p.push(i);
            }
        }
   
        return p;
    }



    word()
    {
        var w = "";
        for (let i in this.madeMoves) {
        w += this.letters[i];
        return w;
        }
    }
}

File: backend/Position.py


class Position:


    def __init__(self, madeMoves,x_,letters_, wordlist):
        self.letters = letters_
        self.x = x_
        self.mm = madeMoves.copy()
        self.pm = self.createPossibleDirections() 
        #max length
        if len(self.mm) < 15:
            self.mm.append(self.x)
            #wordlist.append(self.word())


    def createPossibleDirections(self):
        possibles = []
        if self.x == 0:
            possibles.append(1)
            possibles.append(4)
            possibles.append(5)
        elif self.x == 1:
            possibles.append(0)
            possibles.append(2)
            possibles.append(4)
            possibles.append(5)
            possibles.append(6)
        elif self.x == 2:
            possibles.append(1)
            possibles.append(3)
            possibles.append(5)
            possibles.append(6)
            possibles.append(7)
        elif self.x == 3:
            possibles.append(2)
            possibles.append(6)
            possibles.append(7)
        elif self.x == 4:
            possibles.append(0)
            possibles.append(1)
            possibles.append(5)
            possibles.append(8)
            possibles.append(9)
        elif self.x == 5:
            possibles.append(0)
            possibles.append(1)
            possibles.append(2)
            possibles.append(4)
            possibles.append(6)
            possibles.append(8)
            possibles.append(9)
            possibles.append(10)
        elif self.x == 6:
            possibles.append(1)
            possibles.append(2)
            possibles.append(3)
            possibles.append(5)
            possibles.append(7)
            possibles.append(9)
            possibles.append(10)
            possibles.append(11)
        elif self.x == 7:
            possibles.append(2)
            possibles.append(3)
            possibles.append(6)
            possibles.append(10)
            possibles.append(11)
        elif self.x == 8:
            possibles.append(4)
            possibles.append(5)
            possibles.append(9)
            possibles.append(12)
            possibles.append(13)
        elif self.x == 9:
            possibles.append(4)
            possibles.append(5)
            possibles.append(6)
            possibles.append(8)
            possibles.append(10)
            possibles.append(12)
            possibles.append(13)
            possibles.append(14)
        elif self.x == 10:
            possibles.append(5)
            possibles.append(6)
            possibles.append(7)
            possibles.append(9)
            possibles.append(11)
            possibles.append(13)
            possibles.append(14)
            possibles.append(15)
        elif self.x == 11:
            possibles.append(6)
            possibles.append(7)
            possibles.append(10)
            possibles.append(14)
            possibles.append(15)
        elif self.x == 12:
            possibles.append(8)
            possibles.append(9)
            possibles.append(13)
        elif self.x == 13:
            possibles.append(8)
            possibles.append(9)
            possibles.append(10)
            possibles.append(12)
            possibles.append(14)
        elif self.x == 14:
            possibles.append(9)
            possibles.append(10)
            possibles.append(11)
            possibles.append(13)
            possibles.append(15)
        elif self.x == 15:
            possibles.append(10)
            possibles.append(11)
            possibles.append(14)

        p = []

        for item in possibles:
            if item in self.mm:
                pass
            else:
                p.append(item)
        return p

    def word(self):
        word = ""
        for item in self.mm:
            word += self.letters[item]

        word += "\n"
        return word








